include/swift/SIL/FieldSensitivePrunedLiveness.h

```diff
@@ -1351,9 +1357,17 @@ class FieldSensitiveMultiDefPrunedLiveRange
   void initializeDef(SILNode *node, TypeTreeLeafTypeRange span) {
     assert(Super::isInitialized());
     defs.insert(node, span);
-    auto *block = node->getParentBlock();
-    defBlocks.insert(block, span);
-    initializeDefBlock(block, span);
+    auto defBlock = getDefinedInBlock(node);
+    defBlocks.insert(defBlock, span);
+    initializeDefBlock(defBlock, span);
```

Comment:
I am fine fixing this on main in a forthcoming commit.

I find the logic here hides what is actually happening. First getDefinedInBlock(node) has a special case for try_apply and then we have here another special case below for try_apply. Why not just do something conditional on try_apply here. Then it is clear where all of the special case code is. That being said, I see you used getDefinedInBlock later in the patch, so maybe it makes sense to do this... IDK. Just reads weird to me.

Reply:
Yeah, I went back and forth on whether to use `getDefinedInBlock` here. Maybe it's premature, but I was also trying to think about how to systematically handle these sorts of def-on-edge cases if any more come up; if it does then having `getDefinedInBlock` already be here and used in more than one place seems like a good thing.